api/src/main/java/org/apache/gravitino/rel/expressions/NamedReference.java

```diff
@@ -105,4 +116,46 @@ public String toString() {
       return String.join(".", fieldName);
     }
   }
+
+  /** A {@link NamedReference} that references a metadata field. */
+  final class MetadataField implements NamedReference {
+    /**
+     * The name of the metadata field that contains the partition name. This is used to reference
+     * the partition name in metadata fields.
+     */
+    public static final String PARTITION_NAME_FIELD = "partition_name";
```

Comment:
suggest using `public static final MetadataField PARTITION_NAME_FIELD =xxx`, so that users can use it directly